src/apiCalls.ts

```diff
@@ -14,3 +14,93 @@ export const signupCall = (
 ) => {
   return axios.post("/api/auth/signup", { username, password, email });
 };
+
+// GET USER FROM TOKEN
+
+export const getUserFromToken = (encodedToken: string) =>
+  axios.get("/api/auth/tokenlogin", {
+    headers: { authorization: encodedToken },
+  });
```

Comment:
Liked how you have extracted calls into a function.
Suggestion: try [axios default headers](https://axios-http.com/docs/config_defaults) so you don't need to pass the header for each call just for login/signup when you receive the token set it in axios.